app/models/concerns/digital_object_concerns/digital_object_data/setters/projects.rb

```diff
@@ -6,8 +6,18 @@ module Projects
       extend ActiveSupport::Concern
 
       def set_projects(digital_object_data)
-        return unless digital_object_data.key?('projects')
-        self.projects = Set.new(digital_object_data['projects'].map { |digital_object_data_project|
+        set_primary_project(digital_object_data)
+        set_other_projects(digital_object_data)
+      end
+
+      def set_primary_project(digital_object_data)
+        return unless digital_object_data.key?('primary_project')
+        self.primary_project = Project.find_by(string_key: digital_object_data['primary_project']['string_key'])
```

Comment:
Since we agreed to raise an exception for set_other_projects when an invalid project string key is supplied, I think we should do the same here.